samples/s03-singleAABB/deviceCode.hlsl

```diff
@@ -0,0 +1,86 @@
+// MIT License
+
+// Copyright (c) 2022 Nathan V. Morrical
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "deviceCode.h"
+#include "gprt.h"
+
+struct Payload
+{
+  float3 color;
+};
+
+GPRT_RAYGEN_PROGRAM(simpleRayGen, (RayGenData, record))
+{
+  Payload payload;
+  uint2 pixelID = DispatchRaysIndex().xy;
+  float2 screen = (float2(pixelID) + 
+                  float2(.5f, .5f)) / float2(record.fbSize);
+
+  RayDesc rayDesc;
+  rayDesc.Origin = record.camera.pos;
+  rayDesc.Direction = 
+    normalize(record.camera.dir_00
+    + screen.x * record.camera.dir_du
+    + screen.y * record.camera.dir_dv
+  );
+  rayDesc.TMin = 0.0;
+  rayDesc.TMax = 10000.0;
+  RaytracingAccelerationStructure world = gprt::getAccelHandle(record.world);
+  TraceRay(
+    world, // the tree
+    RAY_FLAG_FORCE_OPAQUE, // ray flags
+    0xff, // instance inclusion mask
+    0, // ray type
+    1, // number of ray types
+    0, // miss type
+    rayDesc, // the ray to trace
+    payload // the payload IO
+  );
+
+  const int fbOfs = pixelID.x + record.fbSize.x * pixelID.y;
+  gprt::store(record.fbPtr, fbOfs, gprt::make_rgba(payload.color));
+}
+
+
+struct Attribute
+{
+  float2 attribs;
+};
+
+GPRT_CLOSEST_HIT_PROGRAM(AABBClosestHit, (AABBGeomData, record), (Payload, payload), (Attribute, attribute))
```

Comment:
Maybe some short comments above these functions explaining what they're doing.

Reply:
we might also do this for the other samples too

Reply:
I added some comments to these samples as well as the previous ones. Hopefully those help explain better what's going on here.